lib/logstash/filters/aggregate.rb

```diff
@@ -317,13 +401,13 @@ def filter(event)
       # retrieve the current aggregate map
       aggregate_maps_element = @@aggregate_maps[task_id]
       
+
       # create aggregate map, if it doesn't exist
       if (aggregate_maps_element.nil?)
         return if @map_action == "update"
         # create new event from previous map, if @push_previous_map_as_event is enabled
         if (@push_previous_map_as_event and !@@aggregate_maps.empty?)
-          previous_map = @@aggregate_maps.shift[1].map
-          event_to_yield = LogStash::Event.new(previous_map)
+          event_to_yield = create_timeout_event(@@aggregate_maps.shift[1].map, task_id)
         end
```

Comment:
Can you keep line `previous_map = @@aggregate_maps.shift[1].map` ?
It allows simply to know what "@@aggregate_maps.shift[1].map" represents.